### DISCUSSION TITLE
Allow for ignoring cache in router#prefetch

### DISCUSSION THREAD

**Discussion by MattIPv4** [vercel/next.js#27730] [Ideas]
### Describe the feature you'd like to request

When using `router#prefetch` calls in the client, there is a private cache ([`sdc`](https://github.com/vercel/next.js/blob/74503f1b8ac61c4cfa625ecfee31373480c54a39/packages/next/shared/lib/router/router.ts#L1621-L1627)) that is not accessible and cannot be bypassed.

This results in there being no way for the client to refresh already cached prefetched page data in the client. I would like for there to be a way to bypass this cache so that I can refresh page data that may have been prefetched on page load and updated in the background by client-side API calls etc.

### Describe the solution you'd like

It would be great to see a new `ignoreCache` boolean added to [`PrefetchOptions`](https://github.com/vercel/next.js/blob/74503f1b8ac61c4cfa625ecfee31373480c54a39/packages/next/shared/lib/router/router.ts#L410-L413) that will bypass the private [`sdc`](https://github.com/vercel/next.js/blob/74503f1b8ac61c4cfa625ecfee31373480c54a39/packages/next/shared/lib/router/router.ts#L1621-L1627) cache and directly re-prefetch the page data.

### Describe alternatives you've considered

Triggering a window refresh or direct window navigation to avoid SPA routing using the cached prefetch data. Incredibly clunky and not what I want to do. After much poking around the Next.js source, I cannot find a way to work around the prefetch cache.

**Comment by SergeyZubatkin** [comment]
What about something like this:
```
import * as nextrouter  from 'next/router'

    if (typeof window !== 'undefined') {
        let staticDataCache = nextrouter.default.router.sdc;
        Object.getOwnPropertyNames(staticDataCache).forEach(function (prop) {
          delete staticDataCache[prop];
        });
      }
```
